composer/callbacks/image_visualizer.py

```
@@ -0,0 +1,189 @@
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair.  The default is 0, which corresponds to any sequence, where the first
+            element is the input. Default: ``0``.
+        target_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the target
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair. The default is 1, which corresponds to any sequence, where the second
+            element is the target. Default: ``1``.
+    """
+
+    def __init__(self,
+                 interval: str = '100ba',
+                 mode: str = 'input',
+                 num_images: int = 8,
+                 input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+                 target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1):
+        self.interval = interval
+        self.mode = mode
+        self.num_images = num_images
+        self.input_key = input_key
+        self.target_key = target_key
+
+        try:
+            import wandb
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='wandb',
+                                                conda_package='wandb',
+                                                conda_channel='conda-forge') from e
+        del wandb  # unused
+
+        # Check that the output mode is valid
+        if self.mode.lower() not in ['input', 'segmentation']:
+            raise ValueError(f'Invalid mode: {mode}')
+
+        # Check that the interval timestring is parsable and convert into time object
+        try:
+            self.interval = Time.from_timestring(interval)
+        except ValueError as error:
+            raise ValueError(f'Invalid time string for parameter interval') from error
```

nit: no need to re-raise the exception; the stack trace should make it obvious that it was the parameter interval that was a bad timestring

---

Cleaned this up